app/src/main/java/com/android/sample/model/recipe/MealDBRecipesRepository.kt

```diff
@@ -0,0 +1,238 @@
+   */
+  private fun parseMealDBJsonToCategory(json: JSONObject): List<String> {
+    val parsedListOfCategories = mutableListOf<String>()
+    val listOfCategories = json.getJSONArray("meals")
+    for (i in 0 until listOfCategories.length()) {
+      val category = listOfCategories.getJSONObject(i)
+      val strCategory = category.getString("strCategory")
+      parsedListOfCategories.add(strCategory)
+    }
+    return parsedListOfCategories
+  }
+
+  /**
+   * Parses the JSON object to a list of thumbnails.
+   *
+   * @param json The JSON object to parse.
+   */
+  private fun parseMealDBJsonToThumbnails(json: JSONObject): List<List<String>> {
+    val parsedListOfThumbnails = mutableListOf<List<String>>()
+    val listOfThumbnails = json.getJSONArray("meals")
+    for (i in 0 until listOfThumbnails.length()) {
+      val thumbnail = listOfThumbnails.getJSONObject(i)
+      val strMeal = thumbnail.getString("strMeal")
+      val idMeal = thumbnail.getString("idMeal")
+      val strMealThumb = thumbnail.getString("strMealThumb")
+      parsedListOfThumbnails.add(listOf(strMeal, idMeal, strMealThumb))
+    }
+    return parsedListOfThumbnails
+  }
+
+  override fun random(
+      nbOfElements: Int,
+      onSuccess: (List<Recipe>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    require(nbOfElements <= MAXIMUM_RECIPES_TO_FETCH_MEAL_DB) {
+      "Please limit the number of recipes to 5"
+    }
+    val listOfRandomRecipes = mutableListOf<Recipe>()
+    // Fetch a random recipe for each element because the API does not support fetching multiple
+    // random recipes at once
+    for (i in 0 until nbOfElements) {
+      getOneRandomRecipe(
+          onSuccess = {
+            listOfRandomRecipes.add(it[0])
+            if (listOfRandomRecipes.size == nbOfElements) {
+              onSuccess(listOfRandomRecipes)
+            }
+          },
+          onFailure = onFailure)
+    }
+  }
+
+  override fun search(mealID: String, onSuccess: (Recipe) -> Unit, onFailure: (Exception) -> Unit) {
+    val url = "$mealDBUrl/lookup.php?i=$mealID"
+    val request =
+        Request.Builder().url(url).header(MEAL_DB_USER_AGENT, MEAL_DB_USER_AGENT_VALUE).build()
+
+    client
+        .newCall(request)
+        .enqueue(
+            object : Callback {
+              override fun onFailure(call: Call, e: IOException) {
+                onFailure(e)
+              }
+
+              override fun onResponse(call: Call, response: Response) {
+                try {
+                  val recipe = parseMealDBJsonToRecipe(JSONObject(response.body!!.string()))
+                  if (recipe.isEmpty()) {
+                    throw JSONException("No recipe found")
+                  }
+                  onSuccess(recipe[0])
+                } catch (e: JSONException) {
+                  onFailure(e)
+                }
+              }
+            })
+  }
+
+  override fun searchByCategory(
+      category: String,
```

Comment:
Could you update the method to accept a list of categories rather than a single category? This will provide greater flexibility and align with the updated requirements.

Reply:
I could do that, do you want it to be an "Or" clause or "And" clause?

Reply:
Since the meal database only supports a single category per recipe, we don’t have the option to use multiple categories. Then an "or" clause is needed.

Reply:
Ok so I tried to do it but it takes more time than I thought it would take, so I suggest to take it as it is right now and as we will change the DB later to bring our recipes to Firestore, I can look if there is a more intuitive way of doing this with Firestore. Is it ok for you ?

Reply:
Ok, no problem! We can stick to just one filter for now.